fbw-common/src/wasm/systems/systems/src/shared/mod.rs

```diff
@@ -727,6 +731,27 @@ impl Average for Pressure {
     }
 }
 
+impl Average for ThermodynamicTemperature {
+    fn average<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = ThermodynamicTemperature>,
+    {
+        let mut sum = 0.0;
+        let mut count: usize = 0;
+
+        for v in iter {
+            sum += v.get::<kelvin>();
+            count += 1;
+        }
+
+        if count > 0 {
+            ThermodynamicTemperature::new::<kelvin>(sum / (count as f64))
+        } else {
+            ThermodynamicTemperature::new::<kelvin>(0.)
+        }
```

Comment:
I think this could use a simple assignment vs if else? 

Also couldn't you logic gate based on ``sum``? I think it is better to manage one state instead of managing the state of 2 variables

Reply:
If you do a direct assignment, what happens if count = 0? You would then have a division by zero case that would panic. You could use something like https://doc.rust-lang.org/stable/std/primitive.i32.html#method.checked_div and then unwrap the `Result`, but I think if else here is actually more readable.